magento/magento2
# Fix for 16345. Adding data to API response for loading products by attribute

### Description (*)

This update done to fix #16345.  Loading product data by API is using `\Magento\Catalog\Model\ProductRepository::get` method and loading products with filter by attributes is using `\Magento\Catalog\Model\ProductRepository::getList`. Product object had no data about `stock_item` on request to load products with filter by attributes. This data was loaded with using plugin `\Magento\CatalogInventory\Model\Plugin\AfterProductLoad`. Same code added to `\Magento\Catalog\Model\ResourceModel\Product\Collection` now, to include  `stock_item` data on request for loading products with filter by attributes.
### Fixed Issues (if relevant)

1. magento/magento2#16345: Getting a product via REST API using a custom attribute as a filter field does NOT return all product data!

### Manual testing scenarios (*)

1. Load product data with using http://{domain.name}/rest/V1/products/{sku}
2. Load products with using http://{domain.name}/rest/all/V1/products/?searchCriteria[filterGroups][0][filters][0][field]={attrributeNameForSameProduct}&searchCriteria[filterGroups][0][filters][0][value]={attrributeValueForSameProduct}
3. Compare product object in response. It should contain same data.

### Questions or comments

I have prepared this update but I am not sure that it is really needed. Stock item data for get product by sku received with using deprecated plugin `\Magento\CatalogInventory\Model\Plugin\AfterProductLoad` and deprecated `\Magento\CatalogInventory\Api\StockRegistryInterface`. And for my opinion it is not bad if response for loading products list contain less data about product then response for loading single product data. Such responses could contain a lot of data about product and request will be heavy. Client application always can make additional requests for loading single product to get all data about this product.
### Contribution checklist (*)
